doc/source/selections.rst

```diff
@@ -0,0 +1,344 @@
+.. -*- coding: utf-8 -*-
+.. _selections:
+
+=======================
+Atom selection language
+=======================
+
+AtomGroups can be created by selecting atoms using the MDAnalysis atom selection language:
+
+  >>> kalp = u.select_atoms("segid KALP")
+
+
+The :meth:`~MDAnalysis.core.groups.AtomGroup.select_atoms` method of a
+:class:`~MDAnalysis.core.groups.AtomGroup` or a
+:class:`~MDAnalysis.core.universe.Universe` returns an
+:class:`~MDAnalysis.core.groups.AtomGroup`. Selections always return an
+:class:`~MDAnalysis.core.groups.AtomGroup` with atoms sorted according to their
+index in the topology. This is to ensure that there are not any duplicates,
+which can happen with complicated selections.
+
+Group subselections using parentheses::
+
+    >>> u.select_atoms("segid DMPC and not (name H* or type OW)")
+    <AtomGroup with 3420 atoms>
+
+Pass in :ref:`named AtomGroups as arguments <preexisting-selections>`::
+
+    >>> sph_6 = u.select_atoms("sphzone 6 protein")
+    >>> u.select_atoms("around 3 group sph_6", sph_6=sph_6)
+    <AtomGroup with 120 atoms>
+
+Turn off :ref:`periodic boundary conditions for geometric keywords <geometric>` with ``periodic=False``::
+
+    >>> u.select_atoms("around 6 protein", periodic=False)
+
+Create :ref:`dynamic UpdatingAtomGroups <dynamic-selections>` with ``updating=True``::
+
+    >>> u.select_atoms("prop x < 5 and prop y < 5 and prop z < 5", updating=True)
+    <AtomGroup with 9 atoms, with selection 'prop x < 5 and prop y < 5 and prop z < 5' on the entire Universe.>
+
+It is possible to export selections for external software
+packages with the help of :ref:`selection-exporters`.
+
+Selection Keywords
+==================
+
+The following describes all selection keywords currently understood by the
+selection parser. The following applies to all selections:
+
+* Keywords are case sensitive.
+* Atoms are automatically sequentially ordered in a resulting selection (see
+  notes below on :ref:`ordered-selections-label` for how to circumvent this if
+  necessary).
+* Selections are parsed left to right and parentheses can be used for
+  grouping.
+* Currently, only "stemming" is implemented as a primitive form of pattern
+  matching: Using the ``*`` character in a string such as ``GL*`` selects
+  all strings that start with "GL" such as "GLU", "GLY", "GLX29", "GLN". Only terminal wildcards (i.e. matching the last part of a name) are currently supported. 
+
+.. note::
+
+    MDAnalysis will ignore everything after the ``*``. ``u.select_atoms("resname *E")`` will not select atoms whose residue name ends in E, but instead select every atom.
+
+
```

Comment:
Is there a way that we only maintain this nice documentation of selections in a single place? If we update it in the code then we mustn't forget to update it here. (Not something to be solved in this PR but generally something to think about: How do we avoid duplicating docs that need to be updated.)